File: analysis/import/SCMLogParser/src/test/java/de/maibornwolff/codecharta/importer/scmlogparser/input/metrics/RelativeCodeChurnTest.java
package de.maibornwolff.codecharta.importer.scmlogparser.input.metrics;


import de.maibornwolff.codecharta.importer.scmlogparser.input.Modification;
import org.junit.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class RelativeCodeChurnTest {
    private static final String FILENAME = "filename";

    @Test
    public void should_have_initial_value_zero() {
        // when
        RelativeCodeChurn metric = new RelativeCodeChurn();

        // then
        assertThat(metric.value()).isEqualTo(0L);
    }

    @Test
    public void should_increase_by_single_modification() {
        // given
        RelativeCodeChurn metric = new RelativeCodeChurn();

        // when
        metric.registerModification(new Modification(FILENAME, 7, 2));

        // then
//        assertThat(metric.value()).isEqualTo(180L);
    }


    @Test
    public void should_increase_by_multiple_modification() {
        // given
        RelativeCodeChurn metric = new RelativeCodeChurn();

        // when
        metric.registerModification(new Modification(FILENAME, 7, 2));
        metric.registerModification(new Modification(FILENAME, 0, 2));
        metric.registerModification(new Modification(FILENAME, 1, 1));
        metric.registerModification(new Modification(FILENAME, 6, 2));

        // then
        // assertThat(metric.value()).isEqualTo(300L);
    }
}

File: analysis/import/SCMLogParser/src/test/kotlin/de/maibornwolff/codecharta/importer/scmlogparser/input/metrics/RelativeCodeChurnTest.kt
package de.maibornwolff.codecharta.importer.scmlogparser.input.metrics


import de.maibornwolff.codecharta.importer.scmlogparser.input.Modification
import org.assertj.core.api.Assertions.assertThat
import org.junit.Test

class RelativeCodeChurnTest {

    private val FILENAME = "filename"

    @Test
    fun should_have_initial_value_zero() {
        // when
        val metric = RelativeCodeChurn()

        // then
        assertThat(metric.value()).isEqualTo(0L)
    }

    @Test
    fun should_increase_by_single_modification() {
        // given
        val metric = RelativeCodeChurn()

        // when
        metric.registerModification(Modification(FILENAME, 7, 2))

        // then
        //        assertThat(metric.value()).isEqualTo(180L);
    }


    @Test
    fun should_increase_by_multiple_modification() {
        // given
        val metric = RelativeCodeChurn()

        // when
        metric.registerModification(Modification(FILENAME, 7, 2))
        metric.registerModification(Modification(FILENAME, 0, 2))
        metric.registerModification(Modification(FILENAME, 1, 1))
        metric.registerModification(Modification(FILENAME, 6, 2))

        // then
        // assertThat(metric.value()).isEqualTo(300L);
    }
}